네브바 구현

GangHub1970 opened the pull request:
## ⭐ Key Changes

<div align="center">
<img width="300px" alt="네브바 이미지" src="https://github.com/user-attachments/assets/2f79cdf3-b0f8-44ca-b4bd-37849a2b79c3"/>
</div>


- 네브바의 기능을 구현했습니다.
  - 네브바에서 사용되는 SVG 아이콘을 추가했습니다.
  - 로고를 만들고 클릭 시 홈(`/`)으로 이동되게 구현했습니다.
  - 네브바의 각 메뉴를 클릭하면 해당되는 페이지로 이동합니다.
  - `숨기기`, `펼치기` 메뉴를 클릭하면 사이드바를 토글할 수 있습니다.

- 네브바를 디자인했습니다.
  - 네브바와 사이드바 모두 `position: fixed`로 위치를 지정했습니다.
  - 현재 경로와 일치하는 네브바 메뉴의 색상과 아이콘을 변경했습니다. ~~이 부분에서 `SearchFill` 아이콘이 아직 없어서 피그마에 올라오면 수정하겠습니다.~~
  - 사이드바가 토글될 때 슬라이드 형식으로 구현했습니다. 피그마에 슬라이드에 대한 언급은 없었지만 슬라이드가 들어가도 좋을 것 같아 넣었습니다..! 피드백 주시면 수정하도록 하겠습니다.

## 🖐️To reviewers

- 기능은 페이지 이동 기능과 사이드바 토글 기능뿐이라 기능 보다는 네브바 선택 시 디자인과 슬라이드바가 토글될 때 디자인을 위주로 피드백 주시면 감사하겠습니다..!!
- 페이지의 경우 검색, 게시물, 마이페이지, 설정 모두 만들었습니다. `/pages`와 `/components` 안에 폴더 네이밍에 대해 피드백 주시면 감사할것 같습니다.
  -  네브바와 사이드바 같은 경우는 `components/@common`에 들어가는게 맞을거같은데 재사용이 안되다보니 `components/layout`으로 생성해보았습니다.
  - 페이지의 경우 `home`, `search`는 채팅, 유저 모두를 다루기 때문에 `pages/common`위치시켰고, `profile`, `setting`는 유저의 정보나 유저의 환경 설정하기 때문에 `pages/user`에 위치시켰습니다.

## 📌issue

close #61

GangHub1970 committed:
feat: 네브바 SVG 아이콘 생성

GangHub1970 committed:
feat: 페이지 생성 및 라우트 연결

- home, search, board, profile, setting 페이지 생성

GangHub1970 committed:
feat: 네브바 기능 구현 및 디자인

GangHub1970 committed:
Merge branch 'develop' into feature/navbar

GangHub1970 committed:
feat: 사이드바 토글 기능 구현

GangHub1970 committed:
design: 검색 Fill 아이콘 추가

iamkjw77 commented on src/components/layout/NavBar/index.tsx:
`@common`과 같이 layout도 전역적으로 사용되는 모듈을 담고 있고, 일반적인 기능을 가진 폴더와 다른 역할을 해서 `@` 식별자를 붙이는 것도 좋을 것 같은데, 어떻게 생각하시나요? @GangHub1970 @itsmesunky

GangHub1970 replied:
저는 좋은것 같습니다!

itsmesunky replied:
저도 좋습니닷

GangHub1970 replied:
수정했습니다!!

iamkjw77 commented on src/components/layout/NavBar/index.tsx:
```diff
@@ -0,0 +1,112 @@
+      </Link>
+      <Styled.NavMenuList>
+        {navMenus.map(({ Icon, ActiveIcon, title, path }) => (
+          <Styled.NavMenu key={uuidv4()}>
```
```suggestion
const id = useId();
...
return(
    ...
  <Styled.NavMenu key={id}>
)     
```
uuid 대신 **useId** 훅을 사용해도 좋을 것 같습니다!

GangHub1970 replied:
`useId`를 사용해서 자식 컴포넌트에 전달하면 `<NavBar>`에서 생성된 id값 하나가 자식 컴포넌트들에게 전달되는 것으로 알고있습니다.

공식문서에서도 프론트에서 만든 데이터의 경우 라이브러리를 사용하는 것을 추천하고 있는것 같아요. 아래 링크에 들어가시면 바로 확인하실 수 있습니다!!
<img width="922" alt="스크린샷 2025-01-08 오전 9 23 14" src="https://github.com/user-attachments/assets/050ee45c-fe55-44e3-8a5a-d0bb33b10399" />

https://ko.react.dev/learn/rendering-lists#where-to-get-your-key

itsmesunky replied:
오호 저도 지원님 피드백을 보고 `useId` 훅과 `uuid`는 어떤 차이가 있나 봤더니
공식 문서에서 `useId`는 리스트 내에서 key 생성 목적의 사용은 지양하고 있네요!

이것도 하나 배워갑니닷 🙇

iamkjw77 replied:
useId는 하나의 컴포넌트의 모든 렌더링에서는 동일한 id를 반환하는군요..!
덕분에 하나 배웠습니다👍👍

iamkjw77 commented on src/components/layout/NavBar/NavBar.styled.ts:
```diff
@@ -0,0 +1,60 @@
+  width: fit-content;
+`;
+
+const NavMenuContentStyle = `
```
```suggestion
import styled, { css } from 'styled-components';

const NavMenuContentStyle = css`
```
styled-components의 css 태그 함수를 사용하면 IDE의 CSS 구문 하이라이팅이나 동적 스타일링(props 기반 스타일링)이 가능해서 css 함수를 사용하는 것도 좋을 것 같습니다☺️

itsmesunky replied:
지원님 피드백을 보고 `styled-components` [공식 문서](https://styled-components.com/docs/api#css)를 확인해봤는데
>A helper function to generate CSS from a template literal with interpolations. You need to use this if you return a template literal with functions inside an interpolation due to how tagged template literals work in JavaScript. If you're interpolating a string you do not need to use this, only if you're interpolating a function.

`css helper` 는 template literal 형태를 반환하는 경우에 사용하라고 하네욥!

지금 당장은 `NavMenuContentStyle` 가 문자열로만 이루어져있지만
확장성, IDE CSS 구문 강조나 오류 검사를 위해 `css` 태그를 사용하는게 좋을 거 같아요〰️

GangHub1970 replied:
말씀해주신대로 CSS 구문 하이라이팅 기능이 추가되는게 좋을 것 같아 css function 을 추가했습니다.

iamkjw77 commented on src/pages/common/home/index.tsx:
pages/의 경우 공통적인 페이지라는 의미가 모호해서, 다른 페이지들과 같이 `pages/home/index.tsx`, `pages/search/index.tsx` 형태로 구성하는 것은 어떨까요? 이렇게 하면 **pages/** 폴더를 열었을 때 어떤 페이지들이 있는지 한눈에 파악할 수 있어 직관적일 것 같습니다. 

이 방식에 대해 어떻게 생각하시나요? @GangHub1970 @itsmesunky

GangHub1970 replied:
저도 공통적인 페이지라는 말이 모호한 것 같아요. 우선 컨벤션이 `/pages/[feature]/home/index.tsx`이런식이라 `/common`에 넣긴했는데 home이랑 search의 경우 하나의 관심사가 정해져 있는게 아니라서 feature 폴더를 만들기가 어려운 것 같아요.

지원님 말씀대로 `pages/home/index.tsx`이렇게 만들어도 좋을 것 같습니다!!

itsmesunky replied:
`pages/home/index.tsx`, `pages/search/index.tsx` 형태로 구성하면
구조가 더 직관적으로 보여서 좋을 것 같습니닷! 😊

GangHub1970 replied:
수정했습니다!!

iamkjw77 reviewed (changes requested):
사이드 바가 토글될 때, 슬라이드 형식으로 애니메이션을 추가하니 훨씬 자연스러운 것 같습니다! 
고생하셨습니다👍

itsmesunky commented on package.json:
```diff
@@ -22,7 +22,8 @@
     "styled-components": "^6.1.13",
-    "styled-reset": "^4.5.2"
+    "styled-reset": "^4.5.2",
+    "uuid": "^11.0.4"
```
`uuid` 라이브러리는 처음 봐서 어디에 쓰이는건가 작업해주신 내용 확인해봤는데
**고유 key 부여**에 사용되는군요..! 하나 배워갑니다 🙇

itsmesunky commented on src/components/layout/NavBar/index.tsx:
```diff
@@ -0,0 +1,112 @@
+          <Styled.CollapseBtn onClick={handleSideBarToggle}>
+            {isSideBarOpen ? (
+              <>
+                <Collapse title="collapse sidebar" width="32px" height="32px" />
```
다른 nav 아이콘 title은 한글로 되어 있는데 collapse/expand title만 영어네욥
통일감 살려서 한글로 바꿔주시면 더 깔끔할 것 같습니다! ✨

GangHub1970 replied:
그러네요..!! `사이드바 숨기기`, `사이드바 펼치기`로 수정했습니다!

GangHub1970 committed:
refactor: 피드백 반영

- NavMenuContentStyle css function 추가
- /components/@layout으로 폴더 네이밍 수정
- /pages/home, /pages/search 로 페이지 컴포넌트 위치 수정

GangHub1970 committed:
style: 사이드바 숨기기/펼치기 아이콘 title 수정

itsmesunky reviewed (approved):
고생하셨습니다 🥳
